src/controllers/preflight.js

```diff
@@ -113,6 +114,20 @@ function PreflightController(ctx, log, env) {
         throw new Error(`No site found for base URL: ${baseURL}`);
       }
 
+      let promiseTokenResponse;
+      if (site.getDeliveryType() === SiteModel.DELIVERY_TYPES.AEM_CS) {
+        try {
+          promiseTokenResponse = await getCSPromiseToken(context);
+          log.info(`Promise token response: ${JSON.stringify(promiseTokenResponse)}`);
+        } catch (e) {
+          log.error(`Failed to get promise token: ${e.message}`);
+          if (e instanceof ErrorWithStatusCode) {
+            return badRequest(e.message);
+          }
+          return createResponse({ message: 'Error getting promise token' }, 500);
```

Comment:
```suggestion
          return internalServerError('Error getting promise token');
```
let's use a dedicated handler